src/daemon/start_mon.sh

```diff
@@ -169,10 +169,11 @@ function start_mon {
       ceph-mon --setuser ceph --setgroup ceph --cluster "${CLUSTER}" -i "${MON_NAME}" --inject-monmap "$MONMAP" --keyring "$MON_KEYRING" --mon-data "$MON_DATA_DIR"
     fi
     if [[ "$CEPH_DAEMON" != demo ]]; then
-      v2v1=$(ceph-conf -c /etc/ceph/"${CLUSTER}".conf 'mon host' | tr ',' '\n' | grep -c "${MON_IP}")
+      MON_IP_NO_BRACKETS=$(echo "$MON_IP" | tr -d '[]')
+      v2v1=$(ceph-conf -c /etc/ceph/"${CLUSTER}".conf 'mon host' | tr ',' '\n' | grep -c "${MON_IP_NO_BRACKETS}")
       # in case of v2+v1 configuration : [v2:xxxx:3300,v1:xxxx:6789]
       if [ "${v2v1}" -eq 2 ]; then
-        timeout 7 ceph "${CLI_OPTS[@]}" mon add "${MON_NAME}" "${MON_IP}" || true
+        timeout 7 ceph "${CLI_OPTS[@]}" mon add "${MON_NAME}" "${MON_IP_NO_BRACKETS}" || true
       # with v2 only : [v2:xxxx:3300]
       else
         timeout 7 ceph "${CLI_OPTS[@]}" mon add "${MON_NAME}" "${MON_IP}":"${MON_PORT}" || true
```

Comment:
I guess we should apply the change when using v2 only (the else statement)

Reply:
no, this is not needed.
I even wonder this might break.

Reply:
```
[root@mon0 ~]# ceph mon add mon1 2620:52:0:880:225:90ff:fefc:1a98:3300
Invalid command: 2620:52:0:880:225:90ff:fefc:1a98:3300 not valid IPv6 address
[root@mon0 ~]# ceph mon add mon1 [2620:52:0:880:225:90ff:fefc:1a98]:3300
adding mon.mon1 at v2:[2620:52:0:880:225:90ff:fefc:1a98]:3300/0
```
```